bin/run_epix

```diff
@@ -49,188 +48,14 @@ def pars_args():
-        dt = epix.times_from_fixed_rate(args.event_rate, number_of_events, args.max_delay)
-        if args.debug:
-            print(f'Fixed event rate of {args.event_rate} Hz')
-    result['t'] = result['t'][:, :] + dt
-
-    if args.debug:
-        print('Generating photons and electrons for events')
-    # Generate quanta:
-    photons, electrons = epix.quanta_from_NEST(epix.awkward_to_flat_numpy(result['ed']),
-                                               epix.awkward_to_flat_numpy(result['nestid']),
-                                               epix.awkward_to_flat_numpy(result['e_field']),
-                                               epix.awkward_to_flat_numpy(result['A']),
-                                               epix.awkward_to_flat_numpy(result['Z']),
-                                               epix.awkward_to_flat_numpy(result['create_S2']),
-                                               density=epix.awkward_to_flat_numpy(result['xe_density']))
-    result['photons'] = epix.reshape_awkward(photons, ak.num(result['ed']))
-    result['electrons'] = epix.reshape_awkward(electrons, ak.num(result['ed']))
-    if args.debug:
-        _ = monitor_time(tnow, 'get quanta.')
-
-    # Reshape instructions:
-    instructions = epix.awkward_to_wfsim_row_style(result)
-
-    # Remove entries with no quanta
-    instructions = instructions[instructions['amp'] > 0]
-    ins_df = pd.DataFrame(instructions)
-
-    if return_df:
-        if args.output_path and not os.path.isdir(args.output_path):
-            os.makedirs(args.output_path)
-
-        output_path_and_name = os.path.join(args.output_path, file_name[:-5] + "_wfsim_instructions.csv")
-        if os.path.isfile(output_path_and_name):
-            warnings.warn("Output file already exists - Overwriting")
-        ins_df.to_csv(output_path_and_name, index=False)
-
-        print('Done')
-        print('Instructions saved to ', output_path_and_name)
-        if args.debug:
-            _ = monitor_time(starttime, 'run epix.')
-
-
-def monitor_time(prev_time, task):
-    t = time.time()
-    print(f'It took {(t - prev_time):.4f} sec to {task}')
-    return t
-
-
-def setup(args):
-    """
-    Function which sets-up configurations. (for strax conversion)
-
-    :return:
-    """
-    # Getting file path and split it into directory and file name:
-    p = args.input_file
-    p = p.split('/')
-    if p[0] == "":
-        p[0] = "/"
-    path = os.path.join(*p[:-1])
-    file_name = p[-1]
-
-    # Init detector volume according to settings and get outer_cylinder
-    # for data reduction of non-relevant interactions.
-    detector = epix.init_detector(args.detector.lower(), args.detector_config)
-    outer_cylinder = getattr(epix.detectors, args.detector.lower())
-    _, outer_cylinder = outer_cylinder()
-    return path, file_name, detector, outer_cylinder
-
-
 if __name__ == "__main__":
-    args = pars_args()
-    main(args, return_df=True)
-
+    args = vars(pars_args())
```

Comment:
Why do we need `vars()` here?

Reply:
Since strax's config is a dictionary we need to be able to acces things like config['variable']. Args' args are accesed as a member (args.var) and this doesn't work with a dictionary. vars(args) casts it into a dictonairy